.azure-pipelines.yml

```diff
@@ -121,3 +122,29 @@ jobs:
   - template: tools/ci/azure/tox_pytest.yml
     parameters:
       directory: tools/wpt/
+
+- job: all_macOS
+  displayName: 'all tests (Safari Technology Preview)'
+  condition: eq(variables['Build.SourceBranch'], 'refs/heads/azure-daily-safari')
+  strategy:
+    parallel: 4 # chosen to make runtime <2h
+  timeoutInMinutes: 360
+  pool:
+    vmImage: 'macOS-10.13'
+  steps:
+  - template: tools/ci/azure/checkout.yml
+  - template: tools/ci/azure/pip_install.yml
+    parameters:
+      packages: virtualenv
+  - template: tools/ci/azure/install_fonts.yml
+  - template: tools/ci/azure/install_certs.yml
+  - template: tools/ci/azure/install_safari.yml
+  - template: tools/ci/azure/update_hosts.yml
+  - template: tools/ci/azure/update_manifest.yml
+  - script: no_proxy='*' ./wpt run --no-manifest-update --no-restart-on-unexpected --no-fail-on-unexpected --this-chunk=$(System.JobPositionInPhase) --total-chunks=$(System.TotalJobsInPhase) --chunk-type hash --log-wptreport $(Build.ArtifactStagingDirectory)/wpt_report_$(System.JobPositionInPhase).json --channel preview safari
+    displayName: 'Run tests'
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish results'
+    inputs:
+      artifactName: 'results'
+    condition: succeededOrFailed()
```

Comment:
@lukebjerring this condition means we'd need to be careful not to use results.zip if the overall check didn't pass. But I thought publishing it even on failure might prove useful for debugging.

Reply:
I was intending on keeping it, it's not just for initial debugging, so I'm not sure what a TODO or comment would say :)

Reply:
IIRC, wptreport.json is only flushed to disk at the *end* of the test suite, so if anything goes wrong in the middle (e.g. `wptrunner` throws an uncaught exception), we'll have a empty file (not as in zero results, but an empty file, which isn't valid JSON). Hence, we probably won't get much value from archiving the output files when the process exits abnormally.